test/tests/functional/pbs_reservations.py

```diff
@@ -42,6 +42,226 @@
 
 class TestReservations(TestFunctional):
 
+    """
+    Various tests to verify behavior of PBS scheduler in handling
+    reservations
+    """
+
+    def create_nodes(self):
+
+        self.server.add_resource("color", "string")
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes('vn', a, 2, self.mom,
+                                  usenatvnode=True, sharednode=False)
+
+        a = {'node_group_key': 'color', 'node_group_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        nodes = self.server.status(NODE)
+        a = {'resources_available.color': 'blue'}
+        self.server.manager(MGR_CMD_SET, NODE, a, nodes[1]['id'])
+        self.__class__._nodes_configured = True
+
+    def submit_resv(self):
+        r = Reservation(TEST_USER)
+        a = {'Resource_List.select': '2:ncpus=1'}
+        r.set_attributes(a)
+        rid = self.server.submit(r)
+        try:
+            a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            d = self.server.expect(RESV, a, rid)
+        except PtlExpectError, e:
+            d = e.rv
+        return d
+
+    def test_no_span_psets(self):
+        self.create_nodes()
+        a = {'do_not_span_psets': '1'}
+        self.server.manager(MGR_CMD_SET, SCHED, a, "")
+        d = self.submit_resv()
+        self.assertTrue(d)
+
+    def test_over_limit(self):
+        self.create_nodes()
+        a = {'max_run_res.ncpus': '[u:' + str(TEST_USER) + '=1]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        d = self.submit_resv()
+        self.assertTrue(d)
+
+    def test_resv_and_job(self):
+        self.create_nodes()
+        a = {'max_run_res.ncpus': '[u:' + str(TEST_USER) + '=3]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        d = self.submit_resv()
+        self.assertTrue(d)
+        j = Job(TEST_USER)
+        j.set_attributes({'Resource_List.walltime': 37 * 3600})
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
+
+    def test_round_robin_reservation(self):
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes('vnode', a, 3, self.mom,
+                                  usenatvnode=True)
+
+        self.scheduler.sched_config[
+            'node_sort_key'] = "\"ncpus HIGH unused\" ALL"
```

Comment:
Why bother setting node_sort_key if you're not checking the order that nodes in the resv_nodes.